### DISCUSSION TITLE
Bitcraze Coordinate System

### DISCUSSION THREAD

**Discussion by tedzaremba** [bitcraze/discussions#1317] [Q&A]
We're a little confused on the coordinate system for the Bitcraze bolt. Looking at [this guide](https://www.bitcraze.io/documentation/system/platform/cf2-coordinate-system/), it appears to be ENU, but we're seeing the quaternion for QY and QZ opposite of our other flight controller's QY and QZ.

**Comment by gemenerik** [comment]
I see your concern. Quaternions can be challenging to interpret, especially when comparing across different systems. Could you clarify your question a little? Are you suggesting your other flight controller should be using the same coordinate system?

**Reply by tedzaremba** [reply]
We would like to transform data on the bolt into data that would match the ENU coordinate system because we're sending data from the bolt via the odometry malvink message to our other flight core running PX4. Below are all the variables we're collecting. Also, I'm not sure if it's okay to mix data from stateEstimate with data from stateEstimateZ as we are doing with mixing the roll, pitch, yaw rate with the position data. 

// State estimate ids
  logVarId_t idx = logGetVarId("stateEstimate", "x");
  logVarId_t idy = logGetVarId("stateEstimate", "y");
  logVarId_t idz = logGetVarId("stateEstimate", "z");

  logVarId_t idvx = logGetVarId("stateEstimate", "vx");
  logVarId_t idvy = logGetVarId("stateEstimate", "vy");
  logVarId_t idvz = logGetVarId("stateEstimate", "vz");

  logVarId_t idqx = logGetVarId("stateEstimate", "qx");
  logVarId_t idqy = logGetVarId("stateEstimate", "qy");
  logVarId_t idqz = logGetVarId("stateEstimate", "qz");
  logVarId_t idqw = logGetVarId("stateEstimate", "qw");

  logVarId_t id_rateRoll = logGetVarId("stateEstimateZ", "rateRoll");
  logVarId_t id_ratePitch = logGetVarId("stateEstimateZ", "ratePitch");
  logVarId_t id_rateYaw = logGetVarId("stateEstimateZ", "rateYaw");

  logVarId_t id_varX = logGetVarId("kalman", "varX");
  logVarId_t id_varY = logGetVarId("kalman", "varY");
  logVarId_t id_varZ = logGetVarId("kalman", "varZ");

  logVarId_t id_varPX = logGetVarId("kalman", "varPX");
  logVarId_t id_varPY = logGetVarId("kalman", "varPY");
  logVarId_t id_varPZ = logGetVarId("kalman", "varPZ");

  logVarId_t id_varD0 = logGetVarId("kalman", "varD0");
  logVarId_t id_varD1 = logGetVarId("kalman", "varD1");
  logVarId_t id_varD2 = logGetVarId("kalman", "varD2");

**Reply by gemenerik** [reply]
`stateEstimateZ` values are compressed log entries, designed to use minimal space while accommodating additional logs in the configuration. The `stateEstimateZ` variables  `rateRoll`, `ratePitch`, and `rateYaw` are stored as int16s, I think it should be fine to use them mixed with raw state estimator data. 

You can verify the requested variables and their units [here](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/api/logs/).

Regarding the coordinate system, are you saying quaternions that should be equal are represented with a flipped sign for QY and QZ on PX4 compared to our firmware?
